micrometer-core/src/main/java/io/micrometer/core/instrument/binder/logging/LogbackMetrics.java

```diff
@@ -182,27 +158,98 @@ public FilterReply decide(Marker marker, Logger logger, Level level, String form
             return FilterReply.NEUTRAL;
         }
 
+        String loggerName = logger.getName();
+
         // cannot use logger.isEnabledFor(level), as it would cause a StackOverflowError by calling this filter again!
         if (level.isGreaterOrEqual(logger.getEffectiveLevel())) {
             switch (level.toInt()) {
                 case Level.ERROR_INT:
-                    errorCounter.increment();
+                    getOrCreateErrorCounter(loggerName).increment();
```

Comment:
"get or create" is already the behavior when registering a meter, so I don't think the Maps add value here unless I'm missing something.